SPECS/mercurial/mercurial.spec

```diff
@@ -51,14 +51,21 @@ popd
 %postun
 /sbin/ldconfig
 
+%define zsh_completions_dir %(pkg-config --variable=completionsdir zsh 2>/dev/null || echo '%{_datadir}/zsh/site-functions')
+
 %files
 %defattr(-,root,root)
 %license COPYING
+/usr/share/bash-completion/completions/hg
```

Comment:
I'm happy to take any feedback on if/how to remove this hardcode. I tried https://github.com/microsoft/CBL-Mariner/pull/7706/commits/18cddafa17fd08ad2e6b23d9ef7af1b8e9e08624#diff-9ccebdf000fd184da23ac8ef0d57f3d8a14ca376632cadab66791cea0970698cR55, but it doesn't evaluate to `/usr/share/bash-completion/completions/` as desired, it fallsback to `%{_sysconfdir}/bash_completion.d` on bb.

![image](https://github.com/microsoft/CBL-Mariner/assets/30801614/75c72ef7-7c1b-49ef-abe1-05e410e40caa)

 It does works as expected in my terminal
![image](https://github.com/microsoft/CBL-Mariner/assets/30801614/1889f74e-a392-4a64-a3c2-16e6bb7c2a29)